### ISSUE TITLE
False positive for no-shadow on typescript enum

### ISSUE THREAD

**Issue by sschoeb** [eslint/eslint#13806]
**Tell us about your environment**
* **ESLint Version: 7.12.1**
* **Node Version: 12.17.0**
* **npm Version: 6.14.4**

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    root: true,
    plugins: [
        "@typescript-eslint"
    ],
    extends: [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking",
    ],
    rules: {
        "no-useless-escape": "off",
        "no-prototype-builtins": "off",
        "@typescript-eslint/no-misused-promises": "off",
        "@typescript-eslint/prefer-regexp-exec": "off",
        "@typescript-eslint/restrict-plus-operands": "off",
        "@typescript-eslint/restrict-template-expressions": "off",
        "@typescript-eslint/unbound-method": "off",
        "@typescript-eslint/no-unused-vars": "off",
        "@typescript-eslint/require-await": "off",
        "@typescript-eslint/no-inferrable-types": ["off", {"ignoreParameters": true,  "ignoreProperties": true}],
        "@typescript-eslint/ban-types": ["error", {
            "types": {
                "object": false,
                "Object": false,
            },
            "extendDefaults": true
        }],
        "no-empty-function": "off",
        "@typescript-eslint/no-empty-function": ["error", {"allow": ["private-constructors"]  }],
        "curly": "error",
        "default-case": "error",
        "eqeqeq": ["error", "smart"],
        "no-eval": "error",
        "no-new-wrappers": "error",
        "no-shadow": "error",
        "no-underscore-dangle": "error",
        "no-trailing-spaces": "error",
        "no-var": "error",
        "@typescript-eslint/type-annotation-spacing": "error",
        "@typescript-eslint/consistent-type-assertions": "error",
        "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
        "@typescript-eslint/no-require-imports": "error",
        "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error"
    }
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
enum Blub2345 {
    A,
    B
}

```

```bash
eslint -c .eslintrc.js "./src/lintTest.ts"
```

**What did you expect to happen?**
No Error

**What actually happened? Please include the actual, raw output from ESLint.**
1:6  error  'Blub2345' is already declared in the upper scope  no-shadow

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by yeonjuan** [comment]
Hi :) @sschoeb 
There is an extension rule `@typescript-eslint/no-shadow` which treats enum.
You should off `no-shadow` and use '@typescript-eslint/no-shadow' instead.

```
rules: {
  'no-shadow': 'off',
  '@typescript-eslint/no-shadow': 'error',
}
```
* see [online-demo](https://yeonjuan.github.io/typescript-eslint-demo/#e2NvZGU6J1xuZW51bSBCbHViMjM0NSB7XG4gICAgQSxcbiAgICBCXG59JyxydWxlczp7J25vLXNoYWRvdyc6J29mZicsJ0B0eXBlc2NyaXB0LWVzbGludC9uby1zaGFkb3cnOidlcnJvcid9LHBhcnNlck9wdGlvbnM6e2VjbWFGZWF0dXJlczp7anN4OmZhbHNlLGdsb2JhbFJldHVybjpmYWxzZX0sZWNtYVZlcnNpb246MjAxOCxwcm9qZWN0OlsnLi90c2NvbmZpZy5qc29uJ10sc291cmNlVHlwZTonc2NyaXB0J319)
